src/test/codegen/export-no-mangle.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![crate_type = "lib"]
+
+mod private {
```

Comment:
Could you please add test cases for these patterns?

``` rust
fn internal() {
    #[no_mangle]
    static EXTERNAL: fn() = internal;
}

static FOO: u32 = {
    #[no_mangle]
    static BAR: &u32 = &FOO;
    0
};

const PRIVATE: () = {
    #[no_mangle]
     fn exported() {}
};
```

I can see myself writing attributes / macros that expand into those patterns.

Reply:
Certainly!